testah-junit/src/main/java/org/testah/framework/dto/TestPlanAnnotationDto.java

```diff
@@ -169,4 +180,52 @@ public String owner() {
         return owner;
     }
 
+    /**
+     * Merge testcase meta data into the testplan for filtering. JUnit4.
+     * @return returns the merged testplan meta.
+     */
+    public TestPlanAnnotationDto mergeWithTestCases() {
```

Comment:
i am not sure about merging the all test case runTypes for a TestPlan together. Suppose a test plan has 3 test cases. The test plan has no annotation but test case 1 has NOT_DEV, test case 2, has NOT_STAGE, test case 3 has NOT_PROD. The test plan ends up with NOT_DEV, NOT_STAGE, NOT_PROD.

Reply:
redesign